frame/nomination-pools/src/lib.rs

```diff
@@ -2157,15 +2157,19 @@ impl<T: Config> Pallet<T> {
 	/// Returns the pending rewards for the specified `member_account`.
 	///
 	/// In the case of error the function returns balance of zero.
-	pub fn pending_rewards(member_account: T::AccountId) -> BalanceOf<T> {
+	pub fn pending_rewards(member_account: T::AccountId) -> Option<BalanceOf<T>> {
 		if let Some(pool_member) = PoolMembers::<T>::get(member_account) {
-			if let Some(reward_pool) = RewardPools::<T>::get(pool_member.pool_id) {
-				return pool_member
-					.pending_rewards(reward_pool.last_recorded_reward_counter())
-					.unwrap_or_default()
+			if let Some((reward_pool, bonded_pool)) = RewardPools::<T>::get(pool_member.pool_id)
+				.zip(BondedPools::<T>::get(pool_member.pool_id))
+			{
+				let current_reward_counter = reward_pool
+					.current_reward_counter(pool_member.pool_id, bonded_pool.points)
+					.ok()?;
+				return Some(pool_member.pending_rewards(current_reward_counter).unwrap_or_default())
```

Comment:
```suggestion
				return Some(pool_member.pending_rewards(current_reward_counter).unwrap_or_else(|| BalanceOf::<T>::max_value()))
```

This would be more correct or?

Reply:
Nope, if for whatever reason we can't fetch their reward, we should show zero, not infinity.